src/ert/scheduler/scheduler.py

```diff
@@ -174,6 +191,34 @@ def add_dispatch_information_to_jobs_file(self) -> None:
         for job in self._jobs.values():
             self._update_jobs_json(job.iens, job.real.run_arg.runpath)
 
+    async def _monitor_and_handle_tasks(
+        self, scheduling_tasks: list[asyncio.Task[None]]
+    ) -> None:
+        pending: Iterable[asyncio.Task[None]] = (
+            list(self._tasks.values()) + scheduling_tasks
+        )
+
+        while True:
+            done, pending = await asyncio.wait(
+                pending,
+                return_when=asyncio.FIRST_COMPLETED,
+            )
+            for task in done:
+                if task.cancelled():
+                    continue
+                if task_exception := task.exception():
+                    logger.error(task_exception)
```

Comment:
Can we include the traceback also?

It is easy enough to find my "1/0" string when testing and seeing "division by zero" in the logs, but there will be more difficult scenarios to come.